week4/MealsharingNET/Controllers/ReservationController.cs

```diff
@@ -0,0 +1,36 @@
+using Microsoft.AspNetCore.Mvc;
+namespace MealsharingNET.Controllers;
+
+[ApiController]
+[Route("Meals")]
+public class ReservationController : ControllerBase
+{
+    private IReservationRepository _repo;
+
+    public ReservationController(IReservationRepository repo)
+    {
+        _repo = repo;
+    }
+
+    [HttpGet("ReservationList")]
+    public List<Reservation> ReservationList()
+    {
+
+        return _repo.ReservationList().ToList();
+    }
+
+    [HttpPost("AddReservation")]
+    public void AddReservation([FromBody] Reservation r)
+    {
+        _repo.CreateReservation(r);
+    }
+
+    [HttpGet("reservationByMealID")]
+    List<Reservation> GetReservationsByMealID(int ID)
```

Comment:
Since the id expected here is a mealId, you should specify the name of the variable as that so it is not confused with reservation id